### PR TITLE
EWPP-27: Fix cache on config change.

### PR DESCRIPTION
**Pull request by drishu**
### Description

Implement cache invalidation for "corporate_default" form display cache tag.

### Change log

- Changed: entity tags to form display tag on presave cache invalidation.

### CODE REVIEW

File: tests/src/Functional/CacheTest.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    $assert = $this->assertSession();
+
+    // Prepare a corporate contact form.
+    $contact_form_id = 'oe_contact_form';
+    $contact_form = ContactForm::create(['id' => $contact_form_id]);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'is_corporate_form', TRUE);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'allow_canonical_url', TRUE);
+    $topic_label = 'Topic label';
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'topic_label', $topic_label);
+    $header = 'this is a test header';
```
Comment (hernani) [inline tests/src/Functional/CacheTest.php:56]: No t() ?

File: oe_contact_forms.module [hunk 2]
```diff
@@ -215,8 +216,15 @@ function _oe_contact_forms_clear_block_cached_definitions(): void {
...
-  Cache::invalidateTags($entity->getCacheTags());
+  // Invalidate form display cache tags if config is changed,
+  // since the EntityFormDisplay is not in storage we have to instantiate here.
+  $entity_form_display = new EntityFormDisplay([
+    'targetEntityType' => 'contact_message',
+    'bundle' => $entity->id(),
+    'mode' => 'corporate_default',
+  ], 'entity_form_display');
+
+  Cache::invalidateTags($entity_form_display->getCacheTags());
```
Comment (hernani) [inline oe_contact_forms.module:227]: $entity->getCacheTags() should not be clear also?

Comment (drishu) [reply oe_contact_forms.module]: My tests where inconclusive so yes, better to be on the safe side

File: tests/src/Functional/CacheTest.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+use Drupal\contact\Entity\ContactForm;
+use Drupal\Tests\BrowserTestBase;
+use Drupal\user\Entity\Role;
+use Drupal\user\RoleInterface;
+
+/**
+ * Tests the corporate contact form rendering under cache.
+ */
+class CacheTest extends BrowserTestBase {
```
Comment (upchuk) [inline tests/src/Functional/CacheTest.php:15]: Why are we making a brand new test? We have other functional tests that where we can ensure this cache gets invalidated...

File: oe_contact_forms.module [hunk 4]
```diff
@@ -215,8 +216,16 @@ function _oe_contact_forms_clear_block_cached_definitions(): void {
  */
 function oe_contact_forms_contact_form_presave(EntityInterface $entity) {
   _oe_contact_forms_clear_block_cached_definitions();
-  // Invalidate cache tags if config is changed.
-  Cache::invalidateTags($entity->getCacheTags());
+  // Invalidate form display cache tags if config is changed,
+  // since the EntityFormDisplay is not in storage we have to instantiate here.
+  $entity_form_display = new EntityFormDisplay([
```
Comment (upchuk) [inline oe_contact_forms.module:221]: is it really need to instantiate a new entity form display to get its cache tags?

File: tests/src/Functional/CacheTest.php (hunk 3 above)
Comment (drishu) [reply tests/src/Functional/CacheTest.php]: moved into MessageFormTest.php

File: oe_contact_forms.module (hunk 4 above)
Comment (drishu) [reply oe_contact_forms.module]: refactored to a simple string

File: oe_contact_forms.module (hunk 2 above)
Comment (upchuk) [reply oe_contact_forms.module]: No need, that gets invalidated on save..

**Pull request merged**